model.py

```diff
@@ -3881,33 +3881,39 @@ def to_search_document(self):
                 dict(name=contributor.name, family_name=contributor.family_name,
                      role=contribution.role))
 
-        # identifier_ids = self.all_identifier_ids()
-        # classifications = Identifier.classifications_for_identifier_ids(
-        #     _db, identifier_ids)
-        # by_scheme_and_term = Counter()
-        # classification_total_weight = 0.0
-        # for c in classifications:
-        #     subject = c.subject
-        #     if subject.type in Subject.uri_lookup:
-        #         scheme = Subject.uri_lookup[subject.type]
-        #         term = subject.name or subject.identifier
-        #         if not term:
-        #             # There's no text to search for.
-        #             continue
-        #         key = (scheme, term)
-        #         by_scheme_and_term[key] += c.weight
-        #         classification_total_weight += c.weight
+        identifier_ids = self.all_identifier_ids()
+        classifications = Identifier.classifications_for_identifier_ids(
+            _db, identifier_ids)
+        by_scheme_and_term = Counter()
+        classification_total_weight = 0.0
+        for c in classifications:
+            subject = c.subject
+            if not subject.type in Subject.TYPES_FOR_SEARCH:
+                # This subject type doesn't have terms that are useful for search
+                continue
+            if subject.type in Subject.uri_lookup:
+                scheme = Subject.uri_lookup[subject.type]
+                term = subject.name or subject.identifier
+                if not term:
+                    # There's no text to search for.
+                    continue
+                term = term.replace("/", " ").replace("--", " ")
```

Comment:
Looks like scrub_identifier doesn't remove the slashes or add spaces, it just adds a slash at the end if there isn't one.

I'm going to leave this here for now but we might need to revisit it later.